### PR TITLE
cosmos metrics

### PR DESCRIPTION
**Pull request by karthik0708**
Dmux can publish system metrics to cosmos

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -38,6 +38,9 @@ RUN groupadd -g ${GID} ${group} && \
     chown -R ${user}:${group} /var/log/go-dmux ; \
     chmod 777 go-dmux
 
+EXPOSE 9999/udp
+EXPOSE 9999/tcp
```
Comment (ikhere) [inline docker/Dockerfile]: is this required to be both udp & tcp?

File: kafka/kafka_source.go [hunk 2]
```diff
@@ -84,7 +85,7 @@ func (k *KafkaSource) Generate(out chan<- interface{}) {
 	if kconf.SASLEnabled {
     		//sarama config plain by default
     		config.Net.SASL.User = kconf.SASLUsername
-    		config.Net.SASL.Password = os.Getenv("KAFKA_CLIENT_USER_PASSWORD")
+    		config.Net.SASL.Password = os.Getenv(kconf.SASLPasswordKey)
```
Comment (ikhere) [inline kafka/kafka_source.go:88]: is this change needed?

File: metrics/send_metrics.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"net/http"
+)
+
+
+func Init() {
+	go send()
+}
+
+func send() {
+	addr := flag.String("listen-address", ":9999", "The address to listen on for HTTP requests.")
```
Comment (ikhere) [inline metrics/send_metrics.go]: let's read this from the existing config file and instead of reading string, let's read port number as integer and build the expected string format in the code.

File: metrics/send_metrics.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+	http.Handle("/metrics", promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{}))
+	promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{})
```
Comment (ikhere) [inline metrics/send_metrics.go]: what is the intent of defining default gatherer? and why do we need second line here?

File: kafka/kafka_source.go (hunk 2 above)
Comment (karthik0708) [reply kafka/kafka_source.go:88]: Created this branch from main. Develop branch and main branch differ here

File: config.go [hunk 5]
```diff
@@ -79,6 +79,7 @@ type DmuxItem struct {
 	Disabled   bool           `json:"disabled`
 	ConnType   ConnectionType `json:"connectionType"`
 	Connection interface{}    `json:connection`
+	MetricPort int			  `json:"metric_port"`
```
Comment (ikhere) [inline config.go]: correct indentation & keep it in line with name, dmuxItems and logging

File: conf.json [hunk 6]
```diff
@@ -5,6 +5,7 @@
       "name": "sample_name",
       "disabled": false,
       "connectionType": "kafka_http",
+      "metric_port": 9999,
```
Comment (ikhere) [inline conf.json]: we can keep this outside `dmuxItems` but in parallel with `name, dmuxItems & logging`

File: main.go [hunk 7]
```diff
@@ -25,6 +26,7 @@ func main() {
 		FilePath: path,
 	}
 	conf := dconf.GetDmuxConf()
+	metrics.Init(conf.DMuxItems[0].MetricPort)
```
Comment (ikhere) [inline main.go]: as commented above, let's keep it separate

File: metrics/send_metrics.go [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+	"strconv"
+)
+
+
+func Init(port int) {
+	go send(port)
+}
+
+func send(port int) {
+	addr := flag.String("listen-address", ":"+strconv.Itoa(port), "The address to listen on for HTTP requests.")
```
Comment (ikhere) [inline metrics/send_metrics.go:17]: correct indentation

**Review by ikhere — COMMENTED** [review]
LGTM. Let's see if `config.go` requires indentation correction.

**Review by ikhere — APPROVED** [review]
